### PR TITLE
Remove date picking in scatter tool

### PR DESCRIPTION
**Pull request by chejennifer**
- Remove date picker and just get latest data for each stat var
- Update how population data is retrieved because need to make sure date of population data for a place matches the date of stat var data for the same place

screen-recording (13)

### CODE REVIEW

**Review by beets — APPROVED** [review]
looks good!

having a separate "point" or "pointcollection" class would be helpful to organize the code. can be done in a followup

File: static/js/tools/scatter2/util.ts [hunk 1]
```diff
@@ -57,26 +65,19 @@ async function getPlacesInNames(
...
-    const sourceSeries = data[dcid];
-    if (_.isEmpty(sourceSeries)) {
-      continue;
-    }
-    sourceSeries["date"] = date;
-  }
-  return data;
+  return axios
+    .get(
+      `/api/stats/collection?parent_place=${parent_place}&child_type=${child_type}${statVarParams}`
```
Comment (shifucun) [inline static/js/tools/scatter2/util.ts:76]: Note this needs to be updated after https://github.com/datacommonsorg/mixer/pull/512 is submitted. Can sync mixer submodule and update this in a follow-up PR

File: static/js/tools/scatter2/chart_loader.tsx [hunk 2]
```diff
@@ -46,35 +46,49 @@ import { StatsVarNode } from "../timeline_util";
...
+  xDate: string;
+  yDate: string;
+  xSource: string;
+  ySource: string;
+  xPop?: number;
+  yPop?: number;
 }
 
-type Cache = Record<string, SourceSeries>;
+type Cache = {
```
Comment (shifucun) [inline static/js/tools/scatter2/chart_loader.tsx:58]: nit: add comment on what the key is (place or statVar)

**Review by shifucun — APPROVED** [review]
Looks good. Much simpler now!

File: static/js/tools/scatter2/chart_loader.tsx [hunk 3]
```diff
@@ -46,35 +46,49 @@ import { StatsVarNode } from "../timeline_util";
...
-          yLog={yVal.log}
-          xPerCapita={xVal.perCapita}
-          yPerCapita={yVal.perCapita}
-          xProvenance={getProvenance(cache, nodeGetStatVar(xVal.statVar))}
-          yProvenance={getProvenance(cache, nodeGetStatVar(yVal.statVar))}
-        />
+      {shouldRenderChart && (
+        <>
+          {_.isEmpty(points) ? (
+            <div className="error-message">Sorry, no data available</div>
```
[the remark this answers is not in the document]
Comment (chejennifer) [reply static/js/tools/scatter2/chart_loader.tsx]: done

File: static/js/tools/scatter2/util.ts (hunk 1 above)
Comment (chejennifer) [reply static/js/tools/scatter2/util.ts:76]: ok, 

> looks good!
> 
> having a separate "point" or "pointcollection" class would be helpful to organize the code. can be done in a followup

There's currently a "point" interface that's being used, how do you want me to use this new class you're suggesting?

Comment (chejennifer) [reply static/js/tools/scatter2/util.ts:76]: ok, good to know, will do that

File: static/js/tools/scatter2/chart_loader.tsx (hunk 2 above)
Comment (chejennifer) [reply static/js/tools/scatter2/chart_loader.tsx:58]: done

**Comment by chejennifer** [comment]
> looks good!
> 
> having a separate "point" or "pointcollection" class would be helpful to organize the code. can be done in a followup

There's currently a "point" interface that I'm using. How would I be using this new class that you're suggesting?

**Comment by beets** [comment]
there's a lot of code there for point manipulation -- factoring it out and upgrading Point to an actual class, with methods, and in a new file would make it cleaner and untangle the nested logic. if there's time...

**Pull request merged**